ansible/roles-infra/infra-osp-project-create/tasks/main.yaml

```diff
@@ -1,79 +1,118 @@
+          instances: "{{ quota_num_instances }}"
+          cores: "{{ quota_num_cores }}"
+          ram: "{{ quota_memory }}" #in MB
+          volumes: "{{ quota_num_volumes }}"
+          gigabytes: "{{ quota_volumes_gigs }}" #volume storage
+          #loadbalancer: #when Octavia is available
+          #pool: #when Octavia is available
+          network: "{{ quota_networks }}"
+          subnet: "{{ quota_subnets }}"
+          router: "{{ quota_routers }}"
+          floatingip: "{{ quota_fip }}"
+          security_group: "{{ quota_sg }}"
+          security_group_rule: "{{ quota_sg_rules }}"
+
+      - name: Create user in new project
+        os_user:
+          state: present
+          name: "{{ guid }}-user"
+          password: "{{ heat_user_password }}"
+          domain: default
+          default_project: "{{ osp_project_name }}"
 
-    - name: Set quotas on new project
-      os_quota:
-        name: "{{ osp_project_name }}"
-        instances: "{{ quota_num_instances }}"
-        cores: "{{ quota_num_cores }}"
-        ram: "{{ quota_memory }}" #in MB
-        volumes: "{{ quota_num_volumes }}"
-        gigabytes: "{{ quota_volumes_gigs }}" #volume storage
-        #loadbalancer: #when Octavia is available
-        #pool: #when Octavia is available
-        network: "{{ quota_networks }}"
-        subnet: "{{ quota_subnets }}"
-        router: "{{ quota_routers }}"
-        floatingip: "{{ quota_fip }}"
-        security_group: "{{ quota_sg }}"
-        security_group_rule: "{{ quota_sg_rules }}"
+      - name: Add member role to user
+        os_user_role:
+          user: "{{ guid }}-user"
+          role: _member_
+          project: "{{ osp_project_name }}"
 
-    - set_fact:
-        heat_user_password_generated: "{{lookup('password', '/dev/null length=12 chars=ascii_letters,digits')}}"
+      - name: Add Swift role to user
+        os_user_role:
+          user: "{{ guid }}-user"
+          role: swiftoperator
+          project: "{{ osp_project_name }}"
+        when: osp_use_swift | default(false) | bool
 
-    - set_fact:
-        heat_user_password: "{{ heat_user_password_provided | default(heat_user_password_generated) }}"
+  - name: Get project info if project already exists
+    when: osp_project_id is not defined
+    block:
+      - name: Get project info
+        # This changes to os_project_info in Ansible 2.9
+        os_project_facts:
+          name: "{{ osp_project_name }}"
+        register: r_osp_project
 
-    - name: Create user in new project
+      - set_fact:
+          osp_project_info: "{{ r_osp_project.ansible_facts.openstack_projects }}"
+
+      - when: osp_project_info | length == 0
+        fail:
+          msg: No project '{{ osp_project_info }}' found
+
+  - name: Add user if sandbox
+    when: osp_create_sandbox
+    block:
+    - name: Create sandbox user in project
       os_user:
         state: present
-        name: "{{ guid }}-user"
+        name: sandbox-{{ guid }}-user
```

Comment:
Why do we need the `sandbox-{{ guid }}-user` for now ? Wasn't it a trick to be able to use just the "special" guid `sandbox-{{guid}}` at the beginning of the PR, before we introduced the `..._member` user ?

i'm OK to have an additional user, but i can't remember why we need it ?

Reply:
No, I think you are correct on this. Removing this and cleaning up the keypair deletion as well.